src/app/(main)/my-voyage/[teamId]/voyage-resources/components/ResourceInput.tsx

```diff
@@ -1,21 +1,95 @@
+import { SubmitHandler, useForm } from "react-hook-form";
+import { useParams } from "next/navigation";
+import * as z from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
 import { LinkIcon } from "@heroicons/react/24/outline";
+import { useAppDispatch } from "@/store/hooks";
+import { validateTextInput } from "@/helpers/form/validateInput";
 import Button from "@/components/Button";
 import TextInput from "@/components/inputs/TextInput";
+import { onOpenModal } from "@/store/features/modal/modalSlice";
+import { addResource } from "@/app/(main)/my-voyage/[teamId]/voyage-resources/resourcesService";
+import useServerAction from "@/hooks/useServerAction";
+import Spinner from "@/components/Spinner";
+
+const validationSchema = z.object({
+  url: validateTextInput({
+    inputName: "Url",
+    required: true,
+    minLen: 1,
+    maxLen: 50,
+  }),
```

Comment:
Add url validation here. I don't think we need anything super fancy. Just a regex to check that it has https:// in it.